eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -120,7 +120,14 @@ public function getContext()
      */
     public function insertContentObject(CreateStruct $struct, $currentVersionNo = 1)
     {
-        $initialLanguageCode = $this->languageHandler->load($struct->initialLanguageId)->languageCode;
+        if (isset($struct->mainLanguageId)) {
+            $initialLanguageId = $struct->mainLanguageId;
```

Comment:
Currently `$initialLanguageCode` is used in two places: 

1. https://github.com/adamwojs/ezpublish-kernel/blob/bdeaef6cf24712fb4fd9ac3af2e3837d29e0a2a7/eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php#L126-L130

2. https://github.com/adamwojs/ezpublish-kernel/blob/bdeaef6cf24712fb4fd9ac3af2e3837d29e0a2a7/eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php#L171-L175

and also `$initialLanguageId` is used two times 😉